### DISCUSSION TITLE
Time Travel Debugging supports

### DISCUSSION THREAD

**Discussion by mochadwi** [orbit-mvi/orbit-mvi#57] [Q&A] [answered]
Been curious about the time-travel debugging approach as shown in: https://github.com/arkivanov/MVIKotlin#debugging-android-application-with-mvikotlin

Does orbit-mvi has public APIs to be consumed (e.g: logging our state)? 

> putting `postSideEffect` for every `intent` seems tedious atm

Or maybe an equivalent approach as workaround? 

We seems unable to find an open issue regarding time-travel debugging atm

**Comment by Rosomack** [comment, marked as answer]
Not right now but it is on our roadmap for sure! We're establishing a good multiplatform base right now so our next features can be developed for both platforms.

**Reply by Praveen-Android** [reply]
Great Framework @Rosomack !! Out of curiosity, checking to see if Time Travel Debugging support is still on your roadmap. Thanks!

**Comment by kenyee** [comment]
Suggestion: add a hook for state changes so we can also add logging in debug mode.  Being able to watch for state changes would help a lot w/ debugging...

**Comment by kenyee** [comment]
Trying to figure out how this would work, but it looks like some sort of hook in ContainerContext would be needed that lets you hook in a time travel singleton plugin that can save/restore state sequences?
